### PR TITLE
Create ReportingControlSample

### PR DESCRIPTION
**Pull request by keith-mcmanus**
Initial submit of Java example for reporting API.  Fully functional, open to critiques for improvement

### CODE REVIEW

File: Java/Marketing-API/MarketingAPISample [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            //encode credentials
+            try{
+               encoding = Base64.getEncoder().encodeToString((serverId +":"+APIKey).getBytes("utf-8") );
+            } catch (Exception d){
+                d.printStackTrace();
+            }
+            //establish connection
+            URL myURL = new URL(serviceURL);
+            connection = (HttpURLConnection)myURL.openConnection();
+                connection.setRequestMethod("DELETE");
```
Comment (BillVolz) [inline Java/Marketing-API/MarketingAPISample]: This line is indented more than the previous.  Was this intentional?

File: Java/Marketing-API/MarketingAPISample [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            contact.put("firstName",firstName);
+            contact.put("lastName",lastName);
+            json = gson.toJson(contact);
+            
+        //with the JSON object created and the authorization crednetials created
+        //we can open the connection and add the contact to the list
+        try{
+        URL myURL = new URL(serviceURL);
+        HttpURLConnection connection = (HttpURLConnection)myURL.openConnection();
+            connection.setRequestMethod("PUT");
```
Comment (BillVolz) [inline Java/Marketing-API/MarketingAPISample]: Same indent issue here.

File: Java/Marketing-API/MarketingAPISample [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            connection.setRequestProperty  ("Authorization","Basic "+encoding);
+            connection.setRequestProperty("Content-Type", "application/json");
+            connection.setUseCaches(false);
+            connection.setDoOutput(true);
+            connection.setDoInput(true);
+            connection.setAllowUserInteraction(false);
+                //writes the entry to your list
```
Comment (BillVolz) [inline Java/Marketing-API/MarketingAPISample]: Extra Indent here too.

File: Java/Marketing-API/MarketingAPISample (hunk 1 above)
Comment (keith-mcmanus) [inline Java/Marketing-API/MarketingAPISample]: Thats just a style I adopted back at Penn State, the indents represent functions related to the object that was created.  Just a style choice I used.  Easy to change.

Comment (BillVolz) [inline Java/Marketing-API/MarketingAPISample]: 100% understand.   Just trying to keep it consistent with the other code in the repository. :)

**Pull request merged**